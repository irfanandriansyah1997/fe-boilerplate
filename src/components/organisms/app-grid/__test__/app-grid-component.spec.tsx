import { fireEvent, render } from '@testing-library/react';

import AppGrid from '..';

describe(`Testing App Grid Testing`, () => {
  it(`Snapshot Testing`, () => {
    const onForceRenderedSpy = jest.fn();
    const onUpdateGridSpy = jest.fn();
    const { getByTestId } = render(
      <AppGrid
        Cell={() => <div>Cell</div>}
        row={3}
        column={4}
        onForceRerender={onForceRenderedSpy}
        onUpdateGrid={onUpdateGridSpy}
      />
    );

    expect(getByTestId(/app-grid/)).toMatchSnapshot();
  });

  it(`Testing Render Grid Component`, () => {
    const onForceRenderedSpy = jest.fn();
    const onUpdateGridSpy = jest.fn();
    const { queryAllByTestId } = render(
      <AppGrid
        Cell={() => <div>Cell</div>}
        row={3}
        column={4}
        onForceRerender={onForceRenderedSpy}
        onUpdateGrid={onUpdateGridSpy}
      />
    );

    const item = queryAllByTestId(/grid-item/);
    const row = queryAllByTestId(/grid-row/);
    const [firstRow] = row;

    expect(row.length).toBe(3);
    expect(firstRow.querySelectorAll(`[data-testid='grid-item']`).length).toBe(
      4
    );
    expect(item.length).toBe(12);
  });

  it(`Testing Simulate Click Button Re-Render & Update Cell`, async () => {
    const onForceRenderedSpy = jest.fn();
    const onUpdateGridSpy = jest.fn();
    const { findByText } = render(
      <AppGrid
        Cell={() => <div>Cell</div>}
        row={3}
        column={4}
        onForceRerender={onForceRenderedSpy}
        onUpdateGrid={onUpdateGridSpy}
      />
    );

    /**
     * Simulate Click Force Re-render
     */
    fireEvent.click(await findByText(/Force Re-render/));
    expect(onUpdateGridSpy).not.toHaveBeenCalled();
    expect(onForceRenderedSpy).toHaveBeenCalled();
    onForceRenderedSpy.mockClear();

    /**
     * Simulate Click Update Cells
     */
    fireEvent.click(await findByText(/Update Cells/));
    expect(onUpdateGridSpy).toHaveBeenCalled();
    expect(onForceRenderedSpy).not.toHaveBeenCalled();
    onForceRenderedSpy.mockClear();
  });
});
