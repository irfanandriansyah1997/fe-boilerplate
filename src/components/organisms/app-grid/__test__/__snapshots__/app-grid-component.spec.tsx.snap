// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Grid Testing Snapshot Testing 1`] = `
<div
  class="app-grid"
  data-testid="app-grid"
>
  <div
    class="app-grid__button-container"
  >
    <button
      type="button"
    >
      <span
        class="material-icons"
      >
        update
      </span>
      Update Cells
    </button>
    <button
      type="button"
    >
      <span
        class="material-icons"
      >
        refresh
      </span>
      Force Re-render
    </button>
  </div>
  <div
    class="grid"
    data-testid="grid-container"
  >
    <div
      class="grid__row"
      data-testid="grid-row"
    >
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
    </div>
    <div
      class="grid__row"
      data-testid="grid-row"
    >
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
    </div>
    <div
      class="grid__row"
      data-testid="grid-row"
    >
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
      <div
        class="grid__column"
        data-testid="grid-item"
      >
        <div>
          Cell
        </div>
      </div>
    </div>
  </div>
</div>
`;
